src/boilerplate/common/services/generic-api_services.mjs

```diff
@@ -0,0 +1,66 @@
+import FUNCTION_NAME from './FUNCTION_NAME.mjs';
+import { startEventFilter, getSiblingPath } from "./common/timber.mjs";
+import fs from "fs";
+import logger from "./common/logger.mjs";
+import { decrypt } from "./common/number-theory.mjs";
+import { searchPartitionedCommitments } from "./common/contract.mjs";
+import web3 from "./common/web3.mjs";
+
+/**
+      Welcome to your zApp's integration test!
+      Depending on how your functions interact and the range of inputs they expect, the below may need to be changed.
+      e.g. Your input contract has two functions, add() and minus(). minus() cannot be called before an initial add() - the compiler won't know this! You'll need to rearrange the below.
+      e.g. The function add() only takes numbers greater than 100. The compiler won't know this, so you'll need to change the call to add() below.
+      The transpiler automatically fills in any ZKP inputs for you and provides some dummy values for the original zol function.
+      NOTE: if any non-secret functions need to be called first, the transpiler won't know! You'll need to add those calls below.
+      NOTE: if you'd like to keep track of your commitments, check out ./common/db/preimage. Remember to delete this file if you'd like to start fresh with a newly deployed contract.
+      */
+const sleep = (ms) => new Promise((r) => setTimeout(r, ms));
+let leafIndex;
+let encryption = {};
+// eslint-disable-next-line func-names
+
+export async function service_FUNCTION_NAME (req, res, next){
+  try {
+    await web3.connect();
+    await new Promise((resolve) => setTimeout(() => resolve(), 3000));
+  } catch (err) {
+    throw new Error(err);
+  }
+	try {
+    await startEventFilter('CONTRACT_NAME');
+    const FUNCTION_SIG;
+    const { tx ,  _RESPONSE_} = await FUNCTION_NAME(FUNCTION_SIG);
+    // prints the tx
+    console.log(tx);
+    res.send({tx, _RESPONSE_});
+    // reassigns leafIndex to the index of the first commitment added by this function
+    if (tx.events.NewLeaves) {
+      leafIndex = tx.events.NewLeaves.returnValues[0];
+      // prints the new leaves (commitments) added by this function call
+      console.log(`Merkle tree event returnValues:`);
+      console.log(tx.events.NewLeaves.returnValues);
+    }
+    if (tx.events.EncryptedData) {
+      if (tx.events.EncryptedData.length) {
+        encryption.msgs = tx.events.EncryptedData[0].returnValues[0];
+        encryption.key = tx.events.EncryptedData[0].returnValues[1];
+        console.log("EncryptedMsgs:");
+        console.log(tx.events.EncryptedData[0].returnValues[0]);
+      } else {
+        encryption.msgs = tx.events.EncryptedData.returnValues[0];
+        encryption.key = tx.events.EncryptedData.returnValues[1];
+        console.log("EncryptedMsgs:");
+        console.log(tx.events.EncryptedData.returnValues);
+      }
+    }
+    await sleep(10);
+  } catch (err) {
+    logger.error(err);
+    res.send({ errors: [err.message] })
+    process.exit(1);
```

Comment:
```suggestion
```
Just one thing - can we remove this? It will crash the APIs and in most cases we dont want the zapp to go down, just to tell you the error so you can try again, which it now does :)